### ISSUE TITLE
How to improve local development server speed (on re-rendering/restarting)?

### ISSUE THREAD

**Issue by TechStacker** [LessWrong2/Lesswrong2#1320]
First of all, thank you so much for making this amazing project open source! 

I was wondering if there's a way to speed up server re-rendering/meteor restart when making changes on my local server? I've timed it to average 44 seconds for each change to take place, e.g. if I add some text to a page and hit save. 

Waiting this much time to see changes is disruptive to the DX and it hurts productivity. 

I'm aware that there are a lot of moving parts in this stack, which makes things like 1-2 second updates a fairy tale, but 30-45 seconds must bother others than just me? 

It makes me think that something might be screwed up in my local environment, as I've seen nobody else report about this (I've searched on lesswrong.com/meta and under Issues here on Github). 

Is there a simple way to disable parts of the app when you e.g. only work on designing & organizing the UI on a specific page? Or is this just one of the downsides of using Meteor and e.g. not being able to use recent versions of Node?  :-)

Besides this, I love everything about this project. Digging through the code is a great learning experience! 

I would appreciate some advice if it's possible to improve local server speed. I'll also gladly write some documentation for other newcomers if I find some solutions on my own :-)

**Comment by Discordius** [comment]
Yeah, I also find this annoying though I am definitely not getting 44 second rebuild times. I did a quick ad-hoc measurement and got an average of 14 seconds. 

I don't think there is a super easy fix to make it much faster. In case you are interested in the details, here is my current analysis of the problem: 

+ Meteor has some infrastructure to only rebuild the client code when you change code that only affects the client, which can bring build-times down some amount (~30%)
+ However, because we do server-side rendering, all frontend code is also imported on the server, causing any change in frontend code to also cause the server-bundle to rebuild
+ In addition to that, we currently don't have App-level code splitting, which means that we always rebuild the full bundle on the client, which takes quite a bit
+ I think our current package architecture might also be adding a bunch to the rebuild times. This [post](https://medium.com/@lucashansen/faster-meteor-reloads-62cca5b56460) has some stuff on that.

**Comment by jimrandomh** [comment]
I just timed it on my desktop and got 18s. Note that Meteor watches for changes and will do a somewhat-incremental update; if you kill and restart it, that will slow it down a lot.
